### PR TITLE
MINOR: [C++] Fix a stale comment and declare a function constexpr

### PR DESCRIPTION
**Pull request by felipecrv**

### CODE REVIEW

File: cpp/src/arrow/type.h [hunk 1]
```diff
@@ -2126,7 +2126,7 @@ Result<std::shared_ptr<Schema>> UnifySchemas(
 
 namespace internal {
 
-static inline bool HasValidityBitmap(Type::type id) {
+inline constexpr bool HasValidityBitmap(Type::type id) {
```
Comment (felipecrv) [inline cpp/src/arrow/type.h:2129]: This was mentioned in another PR of mine, but I'm pushing it here instead to reduce the noise there.

Comment (js8544) [reply cpp/src/arrow/type.h:2129]: constexpr functions are implicitly inline so we don't need the inline keywork here, right?

Comment (felipecrv) [reply cpp/src/arrow/type.h]: Pushed a change.

Comment (js8544) [reply cpp/src/arrow/type.h]: I think the arrow codebase tends to omit inline keyword for constexpr functions, for example in type_trait.h and [this PR](https://github.com/apache/arrow/pull/14241). 
P.S. I don't think the inline specifier has much to do with inline function expansion for modern compilers. It's more for making sure ODR is not violated.

Comment (pitrou) [reply cpp/src/arrow/type.h]: Yes, according to the C++ reference, "A constexpr specifier used in a function declaration implies inline".
https://en.cppreference.com/w/cpp/language/constexpr

**Review by pitrou — APPROVED** [review]
LGTM, thank you @felipecrv

**Pull request merged**